### PR TITLE
llvmPackages_*.llvm: fix building on older CPUs

### LINKED ISSUE #119673: llvmPackages_12.llvm fails to build on wendy
**Describe the bug**
wendy, one of the big-parallel build hosts used by hydra.nixos.org, cannot successfully build llvmPackages_12.llvm. This blocks advancing nixos-unstable.

**To Reproduce**
1. Use wendy to build llvmPackages_12.llvm. Observe [surprising sadness](https://hydra.nixos.org/build/141548680) :(

or, more realistically:

1. Take the attached CMOV16rm-uops.yaml, and run `llvm-exegesis -mcpu=bdver2 -mode=analysis -benchmarks-file=../../CMOV16rm-uops.yaml -analysis-clusters-output-file=- -analysis-clustering-epsilon=0.1 -analysis-inconsistency-epsilon=0.1 -analysis-numpoints=1 -analysis-clustering=naive`

(a reduced commandline that produces the same assertion failure is `llvm-exegesis -mode=analysis -benchmarks-file=../../CMOV16rm-uops.yaml -analysis-clusters-output-file=- -analysis-numpoints=1`)

**Expected behavior**

**Additional context**

The assertion that's tripped by this YAML file is:

```
llvm-exegesis: /home/lukegb/debugit/llvm-project/llvm/include/llvm/MC/MCSchedule.h:348: const llvm::MCSchedClassDesc* llvm::MCSchedModel::getSchedClassDesc(unsigned int) const: Assertion `hasInstrSchedModel() && "No scheduling machine model"' failed.
```

**Notify maintainers**

@lovek323 @7c6f434c @dtzWill @primeos 

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: llvmPackages_12.llvm
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by lukegb**
###### Motivation for this change

This commit patches one of the llvm-exegesis tests to swap out whatever
CPU model happens to be on the build host for bdver2 (AMD Family
15h/Piledriver), which was picked because it looks like that was the
intent of the test author. This provides a more predictable compilation
behaviour when running on older (or possibly even newer!) machines.

One of the machines that is currently part of the NixOS Hydra build
farm, wendy, is using an old AMD Opteron CPU for which LLVM has no
scheduling machine model. This causes one of the tests for llvm-exegesis
to fail, because it segfaults trying to use the machine model to produce
useful analysis results.

Note that this particular test only runs on x86-64 build hosts anyway;
aarch64 isn't affected.

This will contribute towards fixing #119673 for LLVM 9, 10, and 11.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by primeos — APPROVED** [review]
Diff LGTM (s. #119678).

File: pkgs/development/compilers/llvm/exegesis-force-bdver2.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+diff --git a/test/tools/llvm-exegesis/X86/uops-CMOV16rm-noreg.s b/test/tools/llvm-exegesis/X86/uops-CMOV16rm-noreg.s
+index 3fc1f31d54dc..a4c9bdd92131 100644
+--- a/test/tools/llvm-exegesis/X86/uops-CMOV16rm-noreg.s
++++ b/test/tools/llvm-exegesis/X86/uops-CMOV16rm-noreg.s
+@@ -1,5 +1,6 @@
+ # RUN: llvm-exegesis -mode=uops -opcode-name=CMOV16rm  -benchmarks-file=%t.CMOV16rm-uops.yaml
+ # RUN: FileCheck -check-prefixes=CHECK-YAML -input-file=%t.CMOV16rm-uops.yaml %s
++# RUN: sed -i 's,cpu_name:.*,cpu_name: bdver2,g' %t.CMOV16rm-uops.yaml
```
Comment (7c6f434c) [inline pkgs/development/compilers/llvm/exegesis-force-bdver2.patch:8]: Why is this here and not one line higher, by the way?

Comment (lukegb) [reply pkgs/development/compilers/llvm/exegesis-force-bdver2.patch:8]: No particular reason; it could be one line higher. It doesn't affect any of the lines that FileCheck is checking so it doesn't really make much of a difference.

Comment (7c6f434c) [reply pkgs/development/compilers/llvm/exegesis-force-bdver2.patch:8]: Ah I see. Annoying that we cannot just ask `llvm-exegesis` to target `bdver2`…

**Review by 7c6f434c — APPROVED** [review]
Diff looks fine

File: pkgs/development/compilers/llvm/exegesis-force-bdver2.patch (hunk 1 above)
Comment (lukegb) [reply pkgs/development/compilers/llvm/exegesis-force-bdver2.patch:8]: Yeah, I filed an upstream bug; I think that was what the intent of the -mcpu=bdver2 flag in the line below was - looking at the code of llvm-exegesis, it's not actually used at all unless in benchmark mode.

**Comment by lukegb** [comment]
Since this hasn't merged yet, I'm going to abandon it in favour of including the upstream patch.

**Comment by vcunat** [comment]
I can't see it happening yet, so here: https://github.com/NixOS/nixpkgs/pull/122803
